shims/common/src/main/java/org/apache/hadoop/hive/shims/Utils.java

```diff
@@ -172,6 +173,10 @@ public static boolean checkFileSystemXAttrSupport(FileSystem fs) throws IOExcept
       LOG.warn("XAttr won't be preserved since it is not supported for file system: " + fs.getUri());
       return false;
     }
+    catch (AccessControlException e) {
+      LOG.warn(e.getMessage());
+      return false;
+    }
```

Comment:
Isn't catching this wrong? Other systems like HDFS throws ACE for permission issues, & here we will be catching them as well?

Reply:
For Isilon, while calling `fs.getXAttrs`  it throws `org.apache.hadoop.security.AccessControlException` with message `Isilon only supports getXAttrs on /.reserved/raw paths`

should I add additional check also in catch? please suggest

Reply:
this function just checks for xAttr support, if it has access issues then it will fail at a much earlier stage

Reply:
it won't fail, it won't preserve xAttr and replication will go on with DistCp not preserving xAttr, this method was added to check if the FS supports xAttrs or not, if it does support, the xAttrs were always preserved.

The FileSystem description of getXAttr
```
   * @throws UnsupportedOperationException if the operation is unsupported
   *         (default outcome).
   */
  public byte[] getXAttr(Path path, String name) throws IOException {
```
https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L3280-L3283

So, Ideally every HCFS should abide by this, so ignoring ACE ain't an option